tests/test_api_users.py

```diff
@@ -323,4 +352,196 @@ def test_03_create_update_delete_unicode_user(self):
             res = self.app.full_dispatch_request()
             self.assertTrue(res.status_code == 200, res)
             result = res.json.get("result")
-            self.assertTrue(result.get("value"))
\ No newline at end of file
+            self.assertTrue(result.get("value"))
+
+    def test_10_additional_attributes(self):
+        from privacyidea.lib.policy import set_policy, ACTION, SCOPE, delete_policy
+        with self.app.test_request_context('/user/attribute',
+                                           method='POST',
+                                           data={"user": "cornelius@realm1",
+                                                 "key": "newattribute",
+                                                 "value": "newvalue"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 403, res)
+            result = res.json.get("result")
+            self.assertFalse(result.get("status"), res.data)
+            self.assertEqual(result.get("error").get("message"),
+                             "You are not allowed to set this custom user attribute!")
+
+        # Allow to set custom attributes
+        set_policy("custom_attr", scope=SCOPE.ADMIN,
+                   action="{0!s}=:*:*".format(ACTION.SET_USER_ATTRIBUTES))
+
+        with self.app.test_request_context('/user/attribute',
+                                           method='POST',
+                                           data={"user": "cornelius@realm1",
+                                                 "key": "newattribute",
+                                                 "value": "newvalue"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertTrue(result.get("status"), res.data)
+            self.assertTrue(result.get("value") >= 0)
+
+        # Now we verify if the user has the additional attribute:
+        with self.app.test_request_context('/user/attribute',
```

Comment:
I think we should also test that the attribute wasn't set before.

Reply:
Will check and ...

Will do.

Reply:
Done in ae1450be